src/components/Stats/StatsComponent.tsx

```diff
@@ -156,12 +155,24 @@ export class StatsComponent extends React.Component<WidgetProps> {
             StatsComponent.STATS_NAME_MAP.forEach((name, type) => {
                 const index = this.statsData.statistics.findIndex(s => s.statsType === type);
                 if (index >= 0) {
+                    let unitString = "";
+                    if (appStore.activeFrame && appStore.activeFrame.unit) {
+                        const unit = appStore.activeFrame.unit;
+                        if (type === CARTA.StatsType.NumPixels) {
+                            unitString = "pixel(s)";
+                        } else if (type === CARTA.StatsType.SumSq) {
+                            unitString = `(${unit})^2`;
+                        } else {
+                            unitString = unit;
+                        }
+                    }
+
                     const value = this.statsData.statistics[index].value;
-                    const displayValue = (isFinite(value) && type !== CARTA.StatsType.NumPixels) ? value.toExponential(4) : value;
+                    const valueString = isFinite(value) ? `${(type === CARTA.StatsType.NumPixels) ? value : value.toExponential(4)} ${unitString}` : "";
```

Comment:
this line change now prevents `NaN` values from being shown as `NaN`. It should still display `NaN` or `INF` for non-finite values